runtime/hashing/jvm/src/aws/smithy/kotlin/runtime/hashing/Crc32cJVM.kt

```diff
@@ -0,0 +1,664 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.hashing
+
+import java.util.zip.Checksum
+import kotlin.experimental.and
+import kotlin.experimental.xor
+
+public actual class Crc32c : Crc32cBase() {
+    private val md = CRC32CImpl()
+
+    override fun update(input: ByteArray, offset: Int, length: Int): Unit = md.update(input, offset, length)
+
+    override fun digest(): ByteArray {
+        val x = digestValue()
+        reset()
+        return byteArrayOf(
+            ((x shl 24) and 0xffu).toByte(),
+            ((x shl 16) and 0xffu).toByte(),
+            ((x shl 8) and 0xffu).toByte(),
+            (x and 0xffu).toByte(),
+        )
+    }
+
+    override fun digestValue(): UInt = md.value.toUInt()
+
+    override fun reset(): Unit = md.reset()
+}
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Some portions of this file Copyright (c) 2004-2006 Intel Corportation
+ * and licensed under the BSD license.
+ */
+
+/**
+ * A pure-java implementation of the CRC32 checksum that uses
+ * the CRC32-C polynomial, the same polynomial used by iSCSI
+ * and implemented on many Intel chipsets supporting SSE4.2.
+ *
+ * Copied from Hadoop 2.3.6:
+ * https://git-wip-us.apache.org/repos/asf?p=hadoop.git;a=blob_plain;
+ * f=hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/PureJavaCrc32C.java;
+ * hb=2120de588b92b9f22b1cc4188761d6a8c61aa778
+ * <p>
+ * This class is Not ThreadSafe
+ *
+ */
+private class CRC32CImpl : Checksum {
```

Comment:
Is there any reason this can't live in common? 

The only JVM specific thing is inheriting `Checksum` which we don't actually need do we? 

If it lives in common we only have to implement it once. It's possible that on native we may end up using CRT bindings so you could keep the `expect` / `actual` to allow each platform to choose what is best but moving this to common would allow each platform to use this pure kotlin version.